third_party/amd/lib/TritonAMDGPUTransforms/CanonicalizePointers.cpp

```diff
@@ -650,6 +814,174 @@ class ConvertAddPtrOp : public PointerCanonicalizationPattern<tt::AddPtrOp> {
     if (propagateAtrs)
       fatPtrs[nextFatPtr].attributes =
           fatPtrs.at({fatPtrBase, fatPtrOffset}).attributes;
+    return success();
+  }
+
+private:
+  LogicalResult
+  rewriteSmallTensorPtr(tt::AddPtrOp addPtrOp, OneToNOpAdaptor adaptor,
+                        ConversionPatternRewriter &rewriter) const {
+    ValueRange remappedPtr = adaptor.getPtr();
+    ValueRange nonRemappedOffset = adaptor.getOffset();
+
+    assert(remappedPtr.size() == 2 && nonRemappedOffset.size() == 1 &&
+           "expected to be satisified in caller");
```

Comment:
"expected to be set by rewrite infra"

Reply:
update the assert message -- this condition is not satisfied by infra.